syntax = "proto3";

package api;

message GetDeviceProfileRequest {
  string id = 1;
}

message DeviceProfile {
  string id = 1;
  string name = 2;
  string region = 3;
}

message ListDeviceProfilesRequest {
  uint32 limit = 1;
  uint32 offset = 2;
  string search = 3;
  string tenant_id = 4;
}

message ListDeviceProfilesResponse {
  uint32 total_count = 1;
  repeated DeviceProfile result = 2;
}

service DeviceProfileService {
  rpc Get (GetDeviceProfileRequest) returns (DeviceProfile);
  rpc List (ListDeviceProfilesRequest) returns (ListDeviceProfilesResponse);
}